picard/tagger.py

```diff
@@ -842,6 +852,33 @@ def finished(file, result):
             file.set_pending()
             self._acoustid.fingerprint(file, partial(finished, file))
 
+    def extract_and_submit_acousticbrainz_features(self, objs):
+        """Extract AcousticBrainz features and submit them."""
+        if not ab_available():
+            return
+
+        for file in iter_files_from_objects(objs):
+            # Skip unmatched files
+            if not file.can_extract():
+                log.warning("AcousticBrainz requires a MusicBrainz Recording ID, but file does not have it: %s" % file.filename)
+            # And process matched ones
+            else:
+                file.set_pending()
+
+                # Check if file was either already processed or sent to the AcousticBrainz server
+                if file.acousticbrainz_features_file:
+                    ab_extractor_callback(self, file, (file.acousticbrainz_features_file, 0, "Writing results"), False)
+                elif file.acousticbrainz_is_duplicate:
+                    ab_extractor_callback(self, file, (None, 0, "Duplicate"), False)
+                else:
+                    file.acousticbrainz_error = False
+                    # Launch the acousticbrainz on a separate process
+                    log.debug("Extracting AcousticBrainz features from %s" % file.filename)
+                    ab_feature_extraction(self,
```

Comment:
```python
ab_feature_extraction(
    self,
    ...
)
```

Reply:
Should I do the same with calls above?

Reply:
As you wish, I think it will be better as it will shorten lines a bit

Reply:
The error parameter is required by the run_task thing, but AFAIK it is more for failing to run a task than giving a bad result. At least it sounds more logical to me to return a bad result as a result.

Reply:
The error Parameter will hold exceptions that were caused by the run task. So it probably should not be fully ignored but handled

Reply:
The only thing that raises an exception os the version extraction (IndexError for parsing, TimeoutException if its not an AB extractor), but I just catch it and return None.

Reply:
Ok